msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/broker/wpj/TestCase1136625.java

```diff
@@ -0,0 +1,169 @@
+                        .broker(mBroker)
+                        .build();
+
+                new AadPromptHandler(promptHandlerParameters)
+                        .handlePrompt(username, password);
+            }
+        }, TokenRequestTimeout.MEDIUM);
+        authResult.assertSuccess();
+        // Update Active Broker from PROD to RC
+        ((AbstractTestBroker)mBroker).setAppInstaller(new LocalApkInstaller());
+        mBroker.install();
+        // Validate upn, device [id, State, isShared]
+        final String accountUpn = mBrokerHost.getAccountUpn();
+        Assert.assertEquals(accountUpn, username);
+        final String deviceIdAfterAcquireToken = mBrokerHost.obtainDeviceId();
+        Assert.assertEquals(deviceIdAfterAcquireToken, deviceId);
+        final String deviceState = mBrokerHost.getDeviceState();
+        Assert.assertTrue("true".equalsIgnoreCase(deviceState));
+        final boolean isDeviceShared =  mBrokerHost.isDeviceShared();
+        Assert.assertFalse(isDeviceShared);
+        // Install cert
+        mBrokerHost.enableBrowserAccess();
+        //Perform Client TLS, Acquire Token
+        final MsalAuthResult authResultTls = msalSdk.acquireTokenInteractive(getTlsAuthTestParams(), new OnInteractionRequired() {
+            @Override
+            public void handleUserInteraction() {
+                final PromptHandlerParameters promptHandlerParameters = PromptHandlerParameters.builder()
+                        .prompt(PromptParameter.SELECT_ACCOUNT)
+                        .loginHint(mLoginHint)
+                        .sessionExpected(false)
+                        .expectingBrokerAccountChooserActivity(false)
+                        .passwordPageExpected(true)
+                        .broker(mBroker)
+                        .build();
+
+                new TlsPromptHandler(promptHandlerParameters)
+                        .handlePrompt(username, password);
+
+            }
+        }, TokenRequestTimeout.MEDIUM);
+        authResultTls.assertSuccess();
+        final String jwt = authResultTls.getAccessToken();
+        final Map<String, ?> responseItems = jwtParser.parseJWT(jwt);
+        final String jwtDeviceId = (String) responseItems.get("deviceid");
+        Assert.assertEquals(deviceId, jwtDeviceId);
+        // Forward time 1 hour to expire AT
+        TestContext.getTestContext().getTestDevice().getSettings().forwardDeviceTimeForOneDay();
+        // wpj leave
+        final String wpjState = mBrokerHost.wpjLeave();
+        Assert.assertEquals("Device leave successful.", wpjState);
+        final String upnNull = mBrokerHost.getAccountUpn();
+        Assert.assertNull("Device is not joined", upnNull);
+        Thread.sleep(TimeUnit.SECONDS.toMillis(5));
+        try {
+            final boolean deviceDeleted = LabDeviceHelper.deleteDevice(accountUpn, deviceId);
+        } catch (final LabApiException e) {
+            Assert.assertTrue(e.getCode() == 400);
```

Comment:
Why are we looking for 400 here?

Reply:
we are unregistering the device in the previous steps, here we are checking the device has been deleted 
https://github.com/AzureAD/microsoft-authentication-library-for-android/blob/702024c4838e2c6cee32160eecaa07d6f8c1fc09/msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/broker/wpj/TestCase1136625.java#L153-L154

Reply:
So if it was already deleted then Lab API throws 400 response?